File: csharp/TwentyTwo/DayOne.cs
using static AdventOfCode.Application;

namespace AdventOfCode.TwentyTwo;

internal static class DayOne
{
    internal static void Solve()
    {
        var records = new List<int>();
        const int currentSum = 0;
        var input = Application.ReadInput("day01");
        
        ReadRecords(records, currentSum, input);

        var max = records.Max();
        var recordsDesc = records.ToList();
        recordsDesc.Sort();
        recordsDesc.Reverse();

        var top3TotalCalories = recordsDesc[0] + recordsDesc[1] + recordsDesc[2];

        Console.WriteLine("\n --- 2022 Day 1: Calorie Counting ---\n");
        Console.WriteLine($"   The Elf with most calories carried was Elf {records.IndexOf(max)} with {max}");

        Console.WriteLine("   Top 3 Elves carrying the most calories: " +
        $"Elf {records.IndexOf(recordsDesc[0])}, " +
        $"Elf {records.IndexOf(recordsDesc[1])}, " +
        $"Elf {records.IndexOf(recordsDesc[2])} " +
        $"with a combined {top3TotalCalories}\n");
    }

    private static int ReadRecords(ICollection<int> records, int currentSum, string input)
    {
        StringReader inputLines = new(input);

        while (true)
        {
            var line = inputLines.ReadLine();

            if (line != null)
            {
                if (string.IsNullOrWhiteSpace(line))
                {
                    records.Add(currentSum);
                    currentSum = 0;
                }
                else
                {
                    currentSum += int.Parse(line);
                }
            }
            else break;
        }

        return currentSum;
    }
}

File: kotlin/src/main/kotlin/com/kleinreveche/adventofcode/twentytwo/DayOne.kt
package com.kleinreveche.adventofcode.twentytwo

import com.kleinreveche.adventofcode.Solver
import com.kleinreveche.adventofcode.Utils

/** --- Day 1: Calorie Counting --- https://adventofcode.com/2022/day/1 */
class DayOne : Solver {
    override fun solve() {
        val records = parseData()
        val recordsDescending = records.sortedDescending()
        val top3TotalCalories = recordsDescending[0] + recordsDescending[1] + recordsDescending[2]

        println()
        println(" --- 2022 Day 1: Calorie Counting ---\n")
        println("   The Elf with most calories carried was Elf ${records.indexOf(records.max()) + 1} with ${records.max()}")
        println(
            "   Top 3 Elves carrying the most calories: " +
                    "Elf ${records.indexOf(recordsDescending[0])}, " +
                    "Elf ${records.indexOf(recordsDescending[1])}, " +
                    "Elf ${records.indexOf(recordsDescending[2])} " +
                    "with a combined $top3TotalCalories\n"
        )
    }

    override fun parseData(): MutableList<Int> {
        val recordsList = mutableListOf<Int>()
        var currentSum = 0
        val lines = Utils.readInput("twentytwo", "day01")!!.trim().lines()

        lines.forEach { line ->
            if (line.isBlank()) {
                recordsList.add(currentSum)
                currentSum = 0
            } else {
                val number = line.trim().toIntOrNull() ?: 0
                currentSum += number
            }
        }
        recordsList.add(currentSum)
        return recordsList
    }

    override fun parseData(input: Int): Any {
        return 0
    }
}